cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/ReplaceCallPass.kt

```diff
@@ -28,29 +28,30 @@ package de.fraunhofer.aisec.cpg.passes
 import de.fraunhofer.aisec.cpg.TranslationContext
 import de.fraunhofer.aisec.cpg.frontends.Handler
 import de.fraunhofer.aisec.cpg.frontends.HasFunctionalCasts
+import de.fraunhofer.aisec.cpg.frontends.HasFunctionalConstructs
 import de.fraunhofer.aisec.cpg.frontends.Language
 import de.fraunhofer.aisec.cpg.frontends.LanguageFrontend
 import de.fraunhofer.aisec.cpg.graph.*
+import de.fraunhofer.aisec.cpg.graph.declarations.RecordDeclaration
 import de.fraunhofer.aisec.cpg.graph.declarations.TranslationUnitDeclaration
 import de.fraunhofer.aisec.cpg.graph.statements.expressions.*
+import de.fraunhofer.aisec.cpg.graph.types.ObjectType
 import de.fraunhofer.aisec.cpg.graph.types.Type
 import de.fraunhofer.aisec.cpg.helpers.SubgraphWalker
 import de.fraunhofer.aisec.cpg.passes.configuration.DependsOn
 import de.fraunhofer.aisec.cpg.passes.configuration.ExecuteBefore
-import de.fraunhofer.aisec.cpg.passes.configuration.RequiresLanguageTrait
 
 /**
- * If a [Language] has the trait [HasFunctionalCasts], we cannot distinguish between a
- * [CallExpression] and a [CastExpression] during the initial translation. This stems from the fact
- * that we might not know all the types yet. We therefore need to handle them as regular call
- * expression in a [LanguageFrontend] or [Handler] and then later replace them with a
- * [CastExpression], if the [CallExpression.callee] refers to name of a [Type] rather than a
- * function.
+ * If a [Language] has the trait [HasFunctionalCasts] or [HasFunctionalConstructs], we cannot
+ * distinguish between [CallExpression], [CastExpression] or [ConstructExpression] during the
+ * initial translation. This stems from the fact that we might not know all the types yet. We
+ * therefore need to handle them as regular call expression in a [LanguageFrontend] or [Handler] and
+ * then later replace them with a [CastExpression] or [ConstructExpression], if the
+ * [CallExpression.callee] refers to name of a [Type] / [RecordDeclaration] rather than a function.
  */
 @ExecuteBefore(EvaluationOrderGraphPass::class)
 @DependsOn(TypeResolver::class)
-@RequiresLanguageTrait(HasFunctionalCasts::class)
```

Comment:
Are we now omitting this because one of two language traits is necessary now and we have no way to express that?

Reply:
So as agreed we can use a base class named `HasFunctionstyleAmbiguity` here.